service/src/main/java/bio/terra/workspace/common/utils/AwsUtils.java

```diff
@@ -207,6 +222,67 @@ public static AwsCredentialsProvider createWsmCredentialProvider(
         authentication.getGoogleJwtAudience());
   }
 
+  public static Credentials getAssumeServiceRoleCredentials(
```

Comment:
Isn't this method equivalent to calling `createWsmCredentialProvider().resolveCredentials()`?  The duration passed to the "get resource credential" WSM API should not apply to the service credential, just the user credential, so does not need to be passed here.

That method makes use of the SDK's `StsAssumeRoleWithWebIdentityCredentialsProvider` so we don't have to make the `assumeRoleWithWebIdentity` call directly.

Reply:
createWsmCredentialProvider().resolveCredentials() returns 
[software.amazon.awssdk.auth.credentials.AwsCredentials](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/auth/credentials/AwsCredentials.html) , which only includes keyId & accessKey. 

fwiu, we need [software.amazon.awssdk.services.sts.model.Credentials](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/sts/model/Credentials.html) for API authentication, it includes sessionToken, expiration time along with keyId & accessKey

Reply:
Let's leave it as is for now and I'll take an action to create a CRL COW for STS to streamline this later.